pluto/src/vis/diagram/Diagram.tsx

```diff
@@ -126,7 +126,7 @@ const isValidConnection: IsValidConnection = (): boolean => true;
 export interface UseReturn {
   edges: Edge[];
   nodes: Node[];
-  onNodesChange: (nodes: Node[]) => void;
+  onNodesChange: (nodes: Node[], changes: RFNodeChange[]) => void;
```

Comment:
Why was this added?

Reply:
This is not the right way to do that. You're now exposing the internal APIs of react-flow, which the Diagram is built to obfuscate. Why don't you add an extra field to the `Node` item or find a different way to propagate that?